src/server/cli.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2017-2022 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InvalidArgumentError, Option, program } from "commander";
+import { VERSION } from "./version";
+
+function assertIntegerOption(value: string): number {
+  const parsed = parseInt(value, 10);
+  if (isNaN(parsed)) throw new InvalidArgumentError("Must be an integer");
+  return parsed;
+}
+
+const port = new Option("-p, --port <number>", "port number").argParser(assertIntegerOption).default(900); // TODO: DEFAULT_PORT
+const serverRoot = new Option("--server-root <root>", "server root").default(""); // TODO: DEFAULT_ROOT
+const serverHost = new Option("--server-host <host>", "server host").default(null); // TODO: DEFAULT_HOST
+const verbose = new Option("--verbose", "verbose mode").default(false);
+const username = new Option("--username <username>", "username");
+const password = new Option("--password <password>", "password");
+
+function assertCredentials(username: string | undefined, password: string | undefined) {
+  if (password && !username || username && !password) {
+    program.error("You need to pass both username and password");
+  }
+}
+
+program
+  .name("turnilo")
+  .description("Turnilo is a data exploration tool that connects to Druid database")
+  .version(VERSION, "--version");
+
+program
+  .command("run-config")
+  .argument("<file>", "path of config file")
+  .addOption(port)
+  .addOption(serverRoot)
+  .addOption(serverHost)
+  .addOption(username)
+  .addOption(password)
+  .addOption(verbose)
+  .action((configPath, { username, password, serverRoot, serverHost, port, verbose }) => {
+    assertCredentials(username, password);
+    /*
+    1. load yml file as config
+    2. override config with all existing options
+    3. create ServerSettings from 2
+    4. create AppSettings and Sources from 2
+    5. create SettingsManager with 4, logger, verbose option, anchorPath (config path) and initialLoadTimeout (ServerSettings.pageMustLoadTimeout)
+    6. start ./app.ts with ServerSettings and code from ./www.ts
+     */
+    console.log("run config with", configPath, "and options", { port, verbose });
```

Comment:
Why only port and verbose options are logged?

Reply:
That's just crude debug statement. For starters, we need to replace it with logger service.

Reply:
console.log vs. sophisticatedLogger doesn't change anything about logged message, it is a question "what to log" not "how to log" :)